core-c-bridge/src/envconfig.rs

```diff
@@ -0,0 +1,251 @@
+                path: None,
+                data: Some(String::from_utf8_lossy(&d).to_string()),
+            },
+        }
+    }
+}
+
+// Helper functions
+fn parse_config_source(
+    path: *const libc::c_char,
+    data: ByteArrayRef,
+) -> Result<Option<CoreDataSource>, String> {
+    if !path.is_null() {
+        match unsafe { CStr::from_ptr(path) }.to_str() {
+            Ok(path_str) => Ok(Some(CoreDataSource::Path(path_str.to_string()))),
+            Err(e) => Err(format!("Invalid path UTF-8: {e}")),
+        }
+    } else if !data.data.is_null() && data.size > 0 {
+        Ok(Some(CoreDataSource::Data(data.to_vec())))
+    } else {
+        Ok(None)
+    }
+}
+
+fn parse_env_vars(env_vars: ByteArrayRef) -> Result<Option<HashMap<String, String>>, String> {
+    if env_vars.data.is_null() || env_vars.size == 0 {
+        return Ok(None);
+    }
+
+    let env_json = std::str::from_utf8(env_vars.to_slice())
+        .map_err(|e| format!("Invalid env vars UTF-8: {e}"))?;
+
+    serde_json::from_str(env_json)
+        .map(Some)
+        .map_err(|e| format!("Invalid env vars JSON: {e}"))
+}
+
+fn send_result<T: Serialize>(
+    result: Result<T, String>,
+    user_data: *mut libc::c_void,
+    callback: ClientConfigCallback,
+) {
+    match result {
+        Ok(data) => match serde_json::to_vec(&data) {
+            Ok(json_bytes) => {
+                let result = ByteArray::from_vec(json_bytes);
+                unsafe { callback(user_data, result.into_raw(), std::ptr::null()) };
+            }
+            Err(e) => {
+                let err = ByteArray::from_utf8(format!("Failed to serialize: {e}"));
+                unsafe { callback(user_data, std::ptr::null(), err.into_raw()) };
+            }
+        },
+        Err(e) => {
+            let err = ByteArray::from_utf8(e);
+            unsafe { callback(user_data, std::ptr::null(), err.into_raw()) };
+        }
+    }
+}
+
+/// Load all client profiles from given sources
+#[unsafe(no_mangle)]
+pub extern "C" fn temporal_core_client_config_load(
+    path: *const libc::c_char,
+    data: ByteArrayRef,
+    disable_file: bool,
+    config_file_strict: bool,
+    env_vars: ByteArrayRef,
+    user_data: *mut libc::c_void,
+    callback: ClientConfigCallback,
+) {
+    let result = || -> Result<ClientConfig, String> {
+        let config_source = parse_config_source(path, data)?;
+        let env_vars_map = parse_env_vars(env_vars)?;
+
+        let options = LoadClientConfigOptions {
+            config_source: if disable_file { None } else { config_source },
+            config_file_strict,
+        };
+
+        let core_config = envconfig::load_client_config(options, env_vars_map.as_ref())
+            .map_err(|e| e.to_string())?;
+
+        Ok(core_config.into())
+    };
+
+    send_result(result(), user_data, callback);
+}
+
+/// Load a single client profile from given sources with env overrides
+#[unsafe(no_mangle)]
+pub extern "C" fn temporal_core_client_config_profile_load(
```

Comment:
Can you be a bit more specific in docs on what the callback receives on success and make it clear that the user must free callback success/fail?